sw/host/opentitanlib/src/test_utils/bitbanging/i2c.rs

```diff
@@ -202,38 +196,41 @@ pub mod decoder {
     }
 
     impl<const SDA: u8, const SCL: u8> Decoder<SDA, SCL> {
-        /// Returns a sample when a fall clock edge happen.
-        /// The returned sample is taken *before* the falling edge, when SCL is high.
+        /// Loops until the clk transitions to low.
+        /// Returns a symbol (Start|Stop) in case the sda transitions while the clk is high.
         /// The caller must make sure that the clock was high in the previous sample.
-        fn sample_on_fall_edge<I>(samples: &mut Peekable<I>) -> Result<Option<Sample<SDA, SCL>>>
+        fn sample_on_fall_edge<I>(samples: &mut Peekable<I>) -> Result<Option<Symbol>>
```

Comment:
I think this doesn't have to be peekable anymore?

Reply:
It does because of the function `sample_on_raise_edge`

Reply:
A peekable iterator is still an iterator so `&mut I` should work?